Tests/SonarQube.Common.UnitTests/AnalysisPropertyFileProviderTests.cs

```diff
@@ -0,0 +1,206 @@
+        {
+            // Arrange
+            TestLogger logger = new TestLogger();
+            string testDir = TestUtils.CreateTestSpecificFolder(this.TestContext);
+            string invalidFile = CreateFile(testDir, "invalidPropertiesFile.txt", "not a valid XML properties file");
+
+            IList<ArgumentInstance> args = new List<ArgumentInstance>();
+            args.Add(new ArgumentInstance(AnalysisPropertyFileProvider.Descriptor, invalidFile));
+
+            // Act
+            AnalysisPropertyFileProvider provider;
+            bool success = AnalysisPropertyFileProvider.TryCreateProvider(args, testDir, logger, out provider);
+
+            // Assert
+            Assert.IsFalse(success, "Not expecting the properties provider to be created successfully");
+            logger.AssertErrorsLogged(1);
+            logger.AssertSingleErrorExists(invalidFile);
+        }
+
+        #endregion
+
+        #region Private methods
+
+        private static string CreateFile(string path, string fileName, string content)
+        {
+            string fullPath = Path.Combine(path, fileName);
+            File.WriteAllText(fullPath, string.Empty);
+            return fullPath;
+        }
+
+        private static string CreateValidPropertiesFile(string path, string fileName)
+        {
+            string fullPath = Path.Combine(path, fileName);
+
+            AnalysisProperties properties = new AnalysisProperties();
+
+            properties = new AnalysisProperties();
+            properties.Add(new Property() { Id = "foo", Value = "bar" });
+
+            properties.Save(fullPath);
+            return fullPath;
+        }
+
+        private static void AddProperty(IList<Property> properties, string key, string value)
+        {
+            properties.Add(new Property() { Id = key, Value = value });
+        }
+
+        #endregion
+
+        #region Checks
+
+        private static AnalysisPropertyFileProvider InitializeAndCheckIsValid(string defaultPropertiesDirectory, IEnumerable<ArgumentInstance> cmdLineArgs, ILogger logger)
+        {
+            AnalysisPropertyFileProvider provider;
+            bool isValid = AnalysisPropertyFileProvider.TryCreateProvider(cmdLineArgs, defaultPropertiesDirectory, logger, out provider);
+            
+            Assert.IsTrue(isValid, "Expecting the provider to be initialized successfully");
+            return provider;
+        }
+
+        private static void AssertExpectedPropertiesFile(string expectedFilePath, AnalysisPropertyFileProvider actualProvider, TestLogger logger)
+        {
+            Assert.IsNotNull(actualProvider.PropertiesFile, "Properties file object should not be null");
+            Assert.AreEqual(expectedFilePath, actualProvider.PropertiesFile.FilePath, "Properties were not loaded from the expected location");
+        }
+
+        private static void AssertPropertyExists(string key, string expectedValue, IAnalysisPropertyProvider actualProvider)
```

Comment:
@duncanpMS This method is unused, but I indeed would expect it to be used in at least one successful "end-to-end" case